cognite_toolkit/_cdf_tk/commands/dump_resource.py

```diff
@@ -557,6 +564,120 @@ def dump_function_code(self, function: Function, folder: Path) -> None:
+            choices=[
+                Choice(
+                    title=f"{app.name} ({app.creator} - {ms_to_datetime(app.last_updated_time)})", value=app.external_id
+                )
+                for app in sorted(self.apps, key=lambda a: a.name)
+            ],
+        ).ask()
+        if not selected_ids:
+            raise ToolkitValueError("No Streamlit app selected for dumping. Aborting...")
+        return tuple(selected_ids)
+
+    def __iter__(self) -> Iterator[tuple[list[Hashable], CogniteResourceList | None, ResourceLoader, None | str]]:
+        identifier = self.identifier or self._interactive_select()
+        loader = StreamlitLoader.create_loader(self.client)
+        if self.apps:
+            yield [], StreamlitList([app for app in self.apps if app.external_id in identifier]), loader, None
+        else:
+            yield list(identifier), None, loader, None
+
+    def dump_code(self, app: Streamlit, folder: Path, console: Console | None = None) -> None:
+        """Dump the code of a Streamlit app to the specified folder.
+
+        The code is extracted from the JSON content of the app file in CDF.
+
+        Args:
+            app (Streamlit): The Streamlit app whose code is to be dumped.
+            folder (Path): The directory where the app code will be saved.
+            console (Console | None): Optional Rich console for printing warnings.
+        """
+        try:
+            content = self.client.files.download_bytes(external_id=app.external_id)
+        except CogniteAPIError as e:
+            if e.code == 400 and "Files ids not found" in e.message:
+                HighSeverityWarning(
+                    f"The source code for {app.external_id!r} could not be retrieved from CDF."
+                ).print_warning(console=console)
+                return
+            raise
+
+        try:
+            json_content = json.loads(content)
+        except json.JSONDecodeError as e:
+            HighSeverityWarning(
+                f"The JSON content for the Streamlit app {app.external_id!r} is corrupt and could not be extracted. "
+                f"Download file with the same external id manually to remediate. {e!s}"
+            ).print_warning(console=console)
+            return
+
+        app_folder = to_directory_compatible(app.external_id or "unknown_external_id")
+        app_path = folder / app_folder
+        app_path.mkdir(exist_ok=True)
+        if isinstance(json_content.get("requirements"), list):
+            requirements_txt = app_path / "requirements.txt"
+            requirements = json_content["requirements"]
+            if not requirements:
+                HighSeverityWarning(
+                    f"The Streamlit app {app.external_id!r} has a requirements.txt file with no content. Skipping..."
+                ).print_warning()
+            else:
+                requirements_txt.write_text("\n".join(requirements), encoding="utf-8")
```

Comment:
I think `pip` has functionality to do this - and probably better validation

Reply:
https://cognitedata.atlassian.net/browse/CDF-25683